__tests__/token_details.test.js

```diff
@@ -0,0 +1,86 @@
+import HathorWallet from '../src/new/wallet';
+import walletApi from '../src/api/wallet';
+
+class FakeHathorWallet {
+  getTokenDetails(...args) {
+    return HathorWallet.prototype.getTokenDetails.call(this, ...args);
+  }
+}
+
+describe('Get token details', () => {
+  const hathorWallet = new FakeHathorWallet();
+
+  test('Message thrown by token details should be the same received from the API', async () => {
+    expect.assertions(1);
+
+    const walletApiSpy = jest.spyOn(walletApi, 'getGeneralTokenInfo')
```

Comment:
Move the `spyOn` declaration to the describe level, so each test can just make the `.mockImplementationOnce` if needed and the `.mockRestore` can change to a `.mockReset` inside an `afterEach` and have a `.mockRestore` on an `afterAll`

Reply:
Good point! Done on [ada7422](https://github.com/HathorNetwork/hathor-wallet-lib/pull/418/commits/ada7422257722da5dac227eaa44f41c61c83f9e1).